File: Common/Architect.Common/Exceptions/WorksheetCellException.vb
Imports System.Runtime.Serialization

Namespace Exceptions

    <Serializable()>
    Public Class WorksheetCellException
        Inherits Exception
        Implements ISerializable

        Private _rowNumber As Integer
        Private _column As String

        Public Sub New()
            MyBase.New()
        End Sub

        Public Sub New(rowNumber As Integer, column As String, inner As Exception)
            MyBase.New(String.Format("Failed to process row {0} column {1}", rowNumber, column), inner)
            _rowNumber = rowNumber
            _column = column
        End Sub

        Public ReadOnly Property RowNumber() As Integer
            Get
                Return _rowNumber
            End Get
        End Property

        Public ReadOnly Property Column() As String
            Get
                Return _column
            End Get
        End Property

    End Class

End Namespace

File: Kernel/Architect.Utilities.Excel/Exceptions/WorksheetCellException.cs
using System;
using System.Runtime.Serialization;

namespace Architect.Utilities.Excel.Exceptions
{
    [Serializable()]
    public class WorksheetCellException : Exception, ISerializable
    {
        private int _rowNumber;
        private string _column;
        private string _verbose;

        public WorksheetCellException() : base()
        {
        }

        public WorksheetCellException(int rowNumber, string column, Exception inner) : base(string.Format("Failed to process row {0} column {1}", rowNumber, column), inner)
        {
            _rowNumber = rowNumber;
            _column = column;
        }

        public WorksheetCellException(int rowNumber, string column, string verbose, Exception inner) : base(string.Format("Failed to process row {0} column '{2}' ({1})", rowNumber, column, verbose), inner)
        {
            _rowNumber = rowNumber;
            _column = column;
            _verbose = verbose;
        }

        public int RowNumber
        {
            get
            {
                return _rowNumber;
            }
        }

        public string Column
        {
            get
            {
                return _column;
            }
        }
        public string Verbose
        {
            get
            {
                return _verbose;
            }
        }
    }
}
